TypeScript issue with React Native + /// <reference lib="dom"/>

Hi there! I am consuming this library via [Draqula](https://github.com/vadimdemedes/draqula), and came across a thorny TypeScript issue that I'm struggling to work around.

Basically, the problem is: Draqula uses this library under the hood, and naturally exports some of the underlying `ky` types. This works great in the browser + DOM context. However, when working in React Native, the DOM library reference (`/// <reference lib="dom"/>`) in `index.d.ts`  wreaks havoc, because the DefinitelyTyped React Native types have a bunch of duplicate identifiers (like Request, FormData, etc). Because of that reference, despite my TypeScript compiler settings excluding DOM from the list of provided TypeScript libs, it gets pulled in anyway.

I'm cognizant of the fact that mine is a fringe use case - but is there any way I could convince you to _not_ include that reference at the top of the file? If you need it, you can always provide `libs` via `tsconfig.json` or the command line - but if you don't need it, then there's no way to escape it unless it isn't included as a reference. If there exists an alternative, I'd love to try that out as well. Thanks in advance!

Just stumbled upon this one as well, also using Draqula in React Native environment. I've tried just removing the reference, but new errors pop up among others:

```
node_modules/ky/index.d.ts:56:14 - error TS2304: Cannot find name 'URL'.
56  prefixUrl?: URL | string;
                ~~~

node_modules/ky/index.d.ts:120:31 - error TS2304: Cannot find name 'URL'.
120 export type Input = Request | URL | string;
```

I think best (and least hacky for `ky`) course of action here would be to patch `ky` using [patch-package](https://npmjs.com/package/patch-package) by removing `dom` reference and adding missing types to my project's `global.d.ts`.

--- a/index.d.ts
+++ b/index.d.ts
@@ -1,5 +1,3 @@
-/// <reference lib="dom"/>
-
 type Primitive = null | undefined | string | number | boolean | symbol | bigint;
 
 type LiteralUnion<LiteralType extends BaseType, BaseType extends Primitive> =

